packages/router-generator/src/generator.ts

```diff
@@ -80,13 +81,15 @@ async function getRouteNodes(config: Config) {
           let isComponent = routePath?.endsWith('/component')
           let isErrorComponent = routePath?.endsWith('/errorComponent')
           let isPendingComponent = routePath?.endsWith('/pendingComponent')
+          let isNotFoundComponent = routePath?.endsWith('/notFoundComponent')
           let isLoader = routePath?.endsWith('/loader')
 
           ;(
             [
               [isComponent, 'component'],
               [isErrorComponent, 'errorComponent'],
               [isPendingComponent, 'pendingComponent'],
+              [isErrorComponent, 'notFoundComponent'],
```

Comment:
Since we're deprecating the component specific suffixes, we don't need to (and shouldn't) support `notFoundComponent` as a suffix. It should indeed be added to the LazyRoute options though :)